### PR TITLE
Adding a second ConfigFile for the image generation

### PR DESCRIPTION
**Pull request by costingalan**
We need to make sure New-WindowsCloudImage returns a .vhdx image when
it's called as a standalone function, as opposed to being called by
New-WindowsOnlineImage.

Also, this fix ensures New-ProtectedZip is called only once when
is required by New-WindowsOnlineImage.

Signed-off-by: Costin Galan <cgalan@cloudbasesolutions.com>

### CODE REVIEW

File: WinImageBuilder.psm1 [hunk 1]
```diff
@@ -907,7 +907,16 @@ function New-WindowsOnlineImage {
     try {
```
Comment (ader1990) [inline WinImageBuilder.psm1]: commit message: as opposed to* and is required*

File: WinImageBuilder.psm1 [hunk 2]
```diff
@@ -907,7 +907,16 @@ function New-WindowsOnlineImage {
     try {
         $barePath = Get-PathWithoutExtension $windowsImageConfig.image_path
         $virtualDiskPath = $barePath + ".vhdx"
-        New-WindowsCloudImage -ConfigFilePath $ConfigFilePath
+
+        # We need different config files for New-WindowsCloudImage and New-WindowsOnlineImage
+        $offlineConfigFilePath = $ConfigFilePath + ".offline"
+        Copy-Item -Path $ConfigFilePath -Destination $offlineConfigFilePath
+        Set-IniFileValue -Path $offlineConfigFilePath -Key 'image_path' -Value $virtualDiskPath -Section 'DEFAULT'
```
Comment (ader1990) [inline WinImageBuilder.psm1]: line too long

File: WinImageBuilder.psm1 [hunk 3]
```diff
@@ -907,7 +907,16 @@ function New-WindowsOnlineImage {
...
         $barePath = Get-PathWithoutExtension $windowsImageConfig.image_path
         $virtualDiskPath = $barePath + ".vhdx"
-        New-WindowsCloudImage -ConfigFilePath $ConfigFilePath
+
+        # We need different config files for New-WindowsCloudImage and New-WindowsOnlineImage
+        $offlineConfigFilePath = $ConfigFilePath + ".offline"
+        Copy-Item -Path $ConfigFilePath -Destination $offlineConfigFilePath
+        Set-IniFileValue -Path $offlineConfigFilePath -Key 'image_path' -Value $virtualDiskPath -Section 'DEFAULT'
+        if ($windowsImageConfig.zip_password) {
+            Remove-IniFileValue -Path $offlineConfigFilePath -Key 'zip_password' -Section 'DEFAULT'
```
Comment (ader1990) [inline WinImageBuilder.psm1]: line too long

File: WinImageBuilder.psm1 (hunk 1 above)
Comment (costingalan) [inline WinImageBuilder.psm1]: Thanks. Changed them.

File: WinImageBuilder.psm1 (hunk 2 above)
Comment (costingalan) [inline WinImageBuilder.psm1]: Thanks. Fixed it.

File: WinImageBuilder.psm1 (hunk 3 above)
Comment (costingalan) [inline WinImageBuilder.psm1]: Thanks. Fixed it.

**Pull request merged**